static/dbstructure.config.php

```diff
@@ -609,6 +609,39 @@
 			"done_pid_priority_created" => ["done", "pid", "priority", "created"]
 		]
 	],
+	"dcontact" => [
+		"comment" => "Diaspora compatible contacts - used in the Diaspora implementation",
+		"fields" => [
+			"uri-id" => ["type" => "int unsigned", "foreign" => ["item-uri" => "id"], "comment" => "Id of the item-uri table entry that contains the dcontact url"],
+			"addr" => ["type" => "varchar(255)", "comment" => ""],
+			"alias" => ["type" => "varchar(255)", "comment" => ""],
+			"nick" => ["type" => "varchar(255)", "comment" => ""],
+			"name" => ["type" => "varchar(255)", "comment" => ""],
+			"given-name" => ["type" => "varchar(255)", "comment" => ""],
+			"family-name" => ["type" => "varchar(255)", "comment" => ""],
+			"photo" => ["type" => "varchar(255)", "comment" => ""],
+			"photo-medium" => ["type" => "varchar(255)", "comment" => ""],
+			"photo-small" => ["type" => "varchar(255)", "comment" => ""],
+			"batch" => ["type" => "varchar(255)", "comment" => ""],
+			"notify" => ["type" => "varchar(255)", "comment" => ""],
+			"poll" => ["type" => "varchar(255)", "comment" => ""],
+			"subscribe" => ["type" => "varchar(255)", "comment" => ""],
+			"searchable" => ["type" => "boolean", "comment" => ""],
+			"pubkey" => ["type" => "text", "comment" => ""],
+			"baseurl" => ["type" => "varchar(255)", "comment" => "baseurl of the diaspora contact"],
+			"gsid" => ["type" => "int unsigned", "foreign" => ["gserver" => "id", "on delete" => "restrict"], "comment" => "Global Server ID"],
+			"updated" => ["type" => "datetime", "not null" => "1", "default" => DBA::NULL_DATETIME, "comment" => ""],
+			"interacting_count" => ["type" => "int unsigned", "default" => 0, "comment" => "Number of contacts this contact interactes with"],
+			"interacted_count" => ["type" => "int unsigned", "default" => 0, "comment" => "Number of contacts that interacted with this contact"],
+			"post_count" => ["type" => "int unsigned", "default" => 0, "comment" => "Number of posts and comments"],
+		],
+		"indexes" => [
+			"PRIMARY" => ["uri-id"],
+			"addr" => ["addr(32)"],
```

Comment:
Shouldn't we possibly make the `addr` unique?

Reply:
This was from your original table design, I wouldn't have dared question it. In my view it should be unique, but 32 bytes is too short to do so reliably.

Reply:
I know. I only realized later that this can be improved.

Reply:
It's okay, but what should it look like now? `["UNIQUE", "addr(190)"]` ? I don't like it either. Another reference to the `item-uri` table?

Reply:
No, from my side it doesn't belong to the `item-uri` table. I think that some `["UNIQUE", "addr"]` should be fine as well.